unit_scaling/_modules.py

```diff
@@ -181,6 +182,73 @@ def forward(self, input: Tensor) -> Tensor:
         return U.linear_readout(input, self.weight, self.bias, self.constraint)
 
 
+@inherit_docstring(
+    short_description=(
+        "Applies a **unit-scaled** 1D convolution to the incoming data."
+        "\nNote that this layer sets :code:`bias=False` by default."
+    ),
+    add_args=[binary_constraint_docstring],
+)
+class Conv1d(nn.Conv1d):
+    def __init__(
+        self,
+        in_channels: int,
+        out_channels: int,
+        kernel_size: int,
+        stride: int = 1,
+        padding: Union[str, int] = 0,
```

Comment:
This implementation can't handle `str` padding parameter: It errors out if we pass in `padding='valid'` in `test_conv1d()` in `test_modules.py`:
```python
def test_conv1d() -> None:
    batch_size = 2**6
    d_in = 2**6 * 3
    d_out = 2**6 * 5
    kernel_size = 11
    seq_len = 2**6 * 7
    input = randn(batch_size, d_in, seq_len, requires_grad=True)
    model = Conv1d(d_in, d_out, kernel_size, padding='valid', bias=True)
    output = model(input)
```
```python
FAILED unit_scaling/tests/test_modules.py::test_conv1d - TypeError: unsupported operand type(s) for +: 'int' and 'str'
```
That said, `str` padding parameter seems to be a can of worms. Maybe just tighten it to only support `padding: int` for now?

Reply:
Good point - thanks for spotting this. I've just disabled the `str` type for now as you suggest